solr/core/src/java/org/apache/solr/core/CoreContainer.java

```diff
@@ -257,6 +260,7 @@ public CoreLoadFailure(CoreDescriptor cd, Exception loadFailure) {
 
   // initially these are the same to collect the plugin-based listeners during init
   private ClusterEventProducer clusterEventProducer;
+  private PlacementPluginFactory placementPluginFactory;
```

Comment:
I believe we have a synchronization issue here on access to that variable. It is not `final` nor `volatile`, access is not synchronized but it is accessed from multiple threads (the command execution Overseer threads calling `getPlacementPluginFactory()`).

Reply:
Since we always use `DelegatingPlacementPluginFactory` I'm going to create a final instance of this wrapper - and then it will be initialized once the plugins registry is ready.